scss/_root.scss

```
@@ -1,16 +1,53 @@
 :root {
-  // Custom variable values only support SassScript inside `#{}`.
+  // Note: Custom variable values only support SassScript inside `#{}`.
+
+  // Colors
+  //
+  // Generate palettes for full colors, grays, and theme colors.
+
   @each $color, $value in $colors {
     --#{$variable-prefix}#{$color}: #{$value};
   }
 
+  @each $color, $value in $grays {
+    --#{$variable-prefix}gray-#{$color}: #{$value};
+  }
+
   @each $color, $value in $theme-colors {
     --#{$variable-prefix}#{$color}: #{$value};
   }
 
-  // Use `inspect` for lists so that quoted items keep the quotes.
+  @each $color, $value in $theme-colors-rgb {
+    --#{$variable-prefix}#{$color}-rgb: #{$value};
+  }
+
+  --#{$variable-prefix}white-rgb: #{to-rgb($white)};
+  --#{$variable-prefix}black-rgb: #{to-rgb($black)};
+  --#{$variable-prefix}body-rgb: #{to-rgb($body-color)};
+
+  // Fonts
+
+  // Note: Use `inspect` for lists so that quoted items keep the quotes.
   // See https://github.com/sass/sass/issues/2383#issuecomment-336349172
   --#{$variable-prefix}font-sans-serif: #{inspect($font-family-sans-serif)};
   --#{$variable-prefix}font-monospace: #{inspect($font-family-monospace)};
   --#{$variable-prefix}gradient: #{$gradient};
+
+  // Root and body
+  // stylelint-disable custom-property-empty-line-before
+  // scss-docs-start root-body-variables
+  @if $font-size-root != null {
+    --#{$variable-prefix}root-font-size: #{$font-size-root};
+  }
+  --#{$variable-prefix}body-font-family: #{$font-family-base};
+  --#{$variable-prefix}body-font-size: #{$font-size-base};
+  --#{$variable-prefix}body-font-weight: #{$font-weight-base};
+  --#{$variable-prefix}body-line-height: #{$line-height-base};
```

`body-` prefix is really needed for these default font vars?

```suggestion
  --#{$variable-prefix}font-family: #{$font-family-base};
  --#{$variable-prefix}font-size: #{$font-size-base};
  --#{$variable-prefix}font-weight: #{$font-weight-base};
  --#{$variable-prefix}line-height: #{$line-height-base};
```

Imo `--bs-font-family` (for example) is descriptive enough

> not sure about `body-color`

---

I'm sticking with the `body-` infix here on account of having base/body in our Sass variables.

---

Yeah, I imagined @mdo, but don't you think this is a good time to rethink these variable names? I mean, a new API is being created in CSS vars, renaming now is fine, renaming later would be a breaking change 😬

---

I too think that the `body-` prefix can (and probably should) be dropped.

Regarding `--fs`, `--fw` etc I am not so sure. The bundle gzipped size will probably not increase too much an the shorter names are not as intuitive. Furthermore such short names can easily lead to name clashes done the line if e.g. other properties like `font-style` would be used in the future.

---

Regarding the `--bs-body-color` I too am not sure but `root-color` or `text-color` also do not seem much better.

In general I wonder why these properties are set on the body and not on `:root`. I think this would align more with the `-base` suffix they have in SCSS.

---

Dropping prefix or infix is not really an option IMHO since we'd see tons of issues popping out regarding naming interferences with any library out there.

Infix is debatable but I'd recommend to keep it since it'd allow one to use another   interface for their own custom properties, and make those simpler names available and ours recognizable.

---

I am also very much in favor of the `--bs` prefix.

However I think that the `body` conveys some semantic meaning which is simply not true (besides them being set on the body - but they can be used everywhere...). Therefore I think that IFF we want an infix we should use something like `base` (like in sass) or `root` (and set the variables there -which should probably be done regardless)